facade/src/main/java/org/jboss/pnc/facade/providers/EnvironmentProviderImpl.java

```diff
@@ -48,7 +51,9 @@ public EnvironmentProviderImpl(BuildEnvironmentRepository repository, Environmen
     @Override
     @RolesAllowed(SYSTEM_USER)
     public Environment store(Environment restEntity) throws DTOValidationException {
-        return super.store(restEntity);
+        Environment storedEntity = super.store(restEntity);
+        deprecateActiveEnvironmentsWithSameName(storedEntity);
+        return storedEntity;
```

Comment:
I'm missing one thing here. When we deprecate an older version of an env which was an upgrade target for another deprecated env, should we update also the DEPRECATION_UPGRADE value which pointed to the currently deprecated environment to point at the new one? That way we would return the upgrade target right away and save the user from having to walk through the deprecated path iteratively.

Reply:
Or maybe not here, but in the deprecateActiveEnvironmentsWithSameName().